homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -43,7 +44,12 @@
 
 ENTITY_ID_FORMAT = DOMAIN + ".{}"
 
-ALARM_SERVICE_SCHEMA = make_entity_service_schema({vol.Optional(ATTR_CODE): cv.string})
+ALARM_SERVICE_SCHEMA = make_entity_service_schema(
+    {
+        vol.Optional(ATTR_CODE): cv.string,
+        vol.Optional(ATTR_DELAY): cv.positive_int,
```

Comment:
This changes an entity model. Please create an issue in our architecture repo for discussion and approval:
https://developers.home-assistant.io/docs/core/entity#changing-the-entity-model

Reply:
Thanks for the feedback. I wasn't aware of that process.
Link to new item: https://github.com/home-assistant/architecture/issues/458

Reply:
@MartinHjelmare, how do we make progress with this one?

Thanks, Tomer.

Reply:
There needs to be some interest from organization members to discuss this to move it forward.

Reply:
@MartinHjelmare , do you think the manual alarm as is today can replace any physical alarm system? I am not aware of any alarm system which can't be configured to delay the alarm when opening the door but to immediately trigger it when someone break the window. Not having this capability is what takes people to custom alarms provided by 3rd parties. We need to start ramping it up or decide we want to deprecate it and use the other solutions which are out there. I myself moved to use [alarmo](https://github.com/nielsfaber/alarmo) which solved this problem to me.

Reply:
I'm neither for or against your suggestion at the moment. I just don't have time to drive all architecture issues that are created. The members community will need to help get the discussion going.

Reply:
@tomer-w Agreed. 6 months sitting in pending for a feature that is completely backwards compatible (and exists in every physical alarm system) is quite surprising given the velocity this project has. There's been 5 independent releases in the last week!